app/javascript/articles/Feed.jsx

```diff
@@ -1,47 +1,44 @@
-import { h, Component } from 'preact';
+import { h } from 'preact';
+import { useEffect, useState } from 'preact/hooks';
 import PropTypes from 'prop-types';
 
+import useNavigation from '../utilities/hooks/useNavigation';
+
 /* global userData sendHapticMessage showModal buttonFormData renderNewSidebarCount */
 
-export class Feed extends Component {
-  constructor(props) {
-    super(props);
+export const Feed = ({ timeFrame, renderFeed }) => {
+  const { reading_list_ids = [] } = userData(); // eslint-disable-line camelcase
+  const [bookmarkedFeedItems, setBookmarkedFeedItems] = useState(
+    new Set(reading_list_ids),
+  );
 
-    const { reading_list_ids = [] } = userData(); // eslint-disable-line camelcase
+  const [feedItems, setFeedItems] = useState([]);
+  const [podcastEpisodes, setPodcastEpisodes] = useState([]);
 
-    this.state = { bookmarkedFeedItems: new Set(reading_list_ids) };
-  }
+  useEffect(() => {
+    setPodcastEpisodes(getPodcastEpisodes());
+  }, []);
 
-  componentDidMount() {
-    const { timeFrame } = this.props;
+  useEffect(() => {
+    const fetchFeedItems = async () => {
+      const feedItems = await getFeedItems(timeFrame);
```

Comment:
This should be wrapped into a try catch. What should be done with an error on fetch?

Reply:
An option that comes to mind is using the Snackbar to tell the User there was a problem fetching the feed

Reply:
I'll take a look at how the Snackbar is currently used elsewhere, thanks!

Reply:
@fdoxyz I experimented a bit with the snackbar and I'm not sure it 100% fits our needs here. Please don't hesitate if I'm in the wrong here.

If an error happens while fetching the feed, then that error is not temporary like a snackbar item would be. I have the feeling that the message that shows up in such a case should stay there until the page is refreshed, or until the user changes page. A snackbar item will disappear after 5 seconds, which would leave the user in an unknown state.

I thought about using a danger notice instead where the article feed would go:

![Screen Shot 2020-09-30 at 17 56 58](https://user-images.githubusercontent.com/5920450/94744264-ac67a600-0346-11eb-9aa6-4a9bffb5ca3d.png)

What do you think?